scripts/export/export.go

```diff
@@ -0,0 +1,93 @@
+package export
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	app "github.com/cosmos/cosmos-sdk/cmd/gaia/app"
+	"github.com/cosmos/cosmos-sdk/codec"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// GenesisFile defines the Gaia genesis format
+type GenesisFile struct {
+	GenesisTime     string                   `json:"genesis_time"`
+	ChainID         string                   `json:"chain_id"`
+	ConsensusParams *tmtypes.ConsensusParams `json:"consensus_params"`
+	AppHash         string                   `json:"app_hash"`
+	AppState        app.GenesisState         `json:"app_state"`
+}
+
+// NewGenesisFile builds a default GenesisDoc and creates a GenesisFile from it
+func NewGenesisFile(cdc *codec.Codec, path string) (GenesisFile, error) {
+
+	genDoc, err := importGenesis(path)
+	if err != nil {
+		return GenesisFile{}, err
+	}
+
+	var appState app.GenesisState
+	if genDoc.AppState == nil {
+		appState = app.GenesisState{}
+	} else {
+		if err = cdc.UnmarshalJSON(genDoc.AppState, &appState); err != nil {
+			return GenesisFile{}, err
+		}
+	}
+
+	return GenesisFile{
+		GenesisTime:     genDoc.GenesisTime.String(),
+		ChainID:         genDoc.ChainID,
+		ConsensusParams: genDoc.ConsensusParams,
+		AppHash:         genDoc.AppHash.String(),
+		AppState:        appState,
+	}, nil
+}
+
+// ValidateInputs validates each of the parameters used by
```

Comment:
Seems like the godoc isn't finished or indicative of what it does. Perhaps we should call it `ValidateBasic`.

Reply:
updated